packages/@aws-cdk/cdk/lib/cloudformation/fn.ts

```diff
@@ -1,22 +1,301 @@
+   * 2, and the maximum is 10.
+   * @param conditions conditions to AND
+   * @returns an FnCondition token
+   */
+  public static andCondition(...conditions: FnCondition[]): FnCondition {
+    return new FnAnd(...conditions);
+  }
+
+  /**
+   * Compares if two values are equal. Returns true if the two values are equal
+   * or false if they aren't.
+   * @param lhs A value of any type that you want to compare.
+   * @param rhs A value of any type that you want to compare.
+   * @returns an FnCondition token
+   */
+  public static equalsCondition(lhs: any, rhs: any): FnCondition {
+    return new FnEquals(lhs, rhs);
+  }
+
+  /**
+   * Returns one value if the specified condition evaluates to true and another
+   * value if the specified condition evaluates to false. Currently, AWS
+   * CloudFormation supports the ``Fn::If`` intrinsic function in the metadata
+   * attribute, update policy attribute, and property values in the Resources
+   * section and Outputs sections of a template. You can use the AWS::NoValue
+   * pseudo parameter as a return value to remove the corresponding property.
+   * @param condition A reference to a condition in the Conditions section. Use
+   * the condition's name to reference it.
+   * @param valueIfTrue A value to be returned if the specified condition
+   * evaluates to true.
+   * @param valueIfFalse A value to be returned if the specified condition
+   * evaluates to false.
+   * @returns an FnCondition token
+   */
+  public static ifCondition(conditionId: string, valueIfTrue: any, valueIfFalse: any): FnCondition {
+    return new FnIf(conditionId, valueIfTrue, valueIfFalse);
+  }
+
+  /**
+   * Returns true for a condition that evaluates to false or returns false for a
+   * condition that evaluates to true. ``Fn::Not`` acts as a NOT operator.
+   * @param condition A condition such as ``Fn::Equals`` that evaluates to true
+   * or false.
+   * @returns an FnCondition token
+   */
+  public static notCondition(condition: FnCondition): FnCondition {
+    return new FnNot(condition);
+  }
+
+  /**
+   * Returns true if any one of the specified conditions evaluate to true, or
+   * returns false if all of the conditions evaluates to false. ``Fn::Or`` acts
+   * as an OR operator. The minimum number of conditions that you can include is
+   * 2, and the maximum is 10.
+   * @param conditions conditions that evaluates to true or false.
+   * @returns an FnCondition token
+   */
+  public static orCondition(...conditions: FnCondition[]): FnCondition {
+    return new FnOr(...conditions);
+  }
+
+  /**
+   * Returns true if a specified string matches at least one value in a list of
+   * strings.
+   * @param listOfStrings A list of strings, such as "A", "B", "C".
+   * @param value A string, such as "A", that you want to compare against a list of strings.
+   * @returns an FnCondition token
+   */
+  public static containsCondition(listOfStrings: string[], value: string): FnCondition {
```

Comment:
I don't like the `Condition` suffix everywhere. It looks mighty redundant. In the case of `containsConditions` here, the name even would imply I'm checking if some array contains a `FnCondition` reference, which is not what it's doing.

I would at the very least remove the `Condition` suffix, and maybe ideally move this into a `Condition` class (`Condition.equals(val, val)`, `Condition.contains(arr, val)`, ...)

Reply:
Some suffix/prefix is needed for two reasons:
- Many of the condition functions use names that are reserved words (such as "if", "and", "or").
- It should be possible for users to easily distinguish between condition functions and normal functions.

I agree about "containsCondition" implying the wrong thing.

Alternative options:
- A prefix "condition": `conditionContains`, `conditionAnd`, etc
- A postfix "cond": `containsCond`, `andCond`
- A prefix "cond": `condContains`, `condIf`

Any other ideas?

Reply:
I prefer going for a prefix (since we don't want to go full on namespace/class, that's what's closest), preferably `condition` (the short-hand doesn't improve much of anything).